app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -295,13 +295,42 @@ private void startPokeFly() {
         Intent intent = Pokefly.createIntent(this, trainerLevel, statusBarHeight, batterySaver);
         startService(intent);
 
-        pokeFlyRunning = true;
-
         if (settings.shouldLaunchPokemonGo()) {
             openPokemonGoApp();
         }
     }
 
+    private boolean isPokeFlyRunning() {
+        ActivityManager manager = (ActivityManager) getSystemService(Context.ACTIVITY_SERVICE);
+        for (ActivityManager.RunningServiceInfo serviceInfo : manager.getRunningServices(Integer.MAX_VALUE)) {
+            if (Pokefly.class.getName().equals(serviceInfo.service.getClassName())) {
+                return true;
+            }
+        }
```

Comment:
This seems like an expensive and inefficient way to check if the service is running. This function should be there, but I think it might actually be better to have Pokefly set/unset some flag. Who knows how many services are running on a users phone?!